packages/eslint-plugin/src/internal.js

```diff
@@ -0,0 +1,26 @@
+function shouldRegisterInternal() {
+  try {
+    const hasNxEslintPlugin = require.resolve('@nx/eslint-plugin');
+    return Boolean(hasNxEslintPlugin);
+  } catch (err) {
+    return false;
+  }
+}
+
+const shouldRegister = shouldRegisterInternal();
+
+/**
+ * @internal
+ *
+ * this will be removed after https://github.com/microsoft/fluentui/issues/30332
+ *
+ * expands this with rulesets/overrides that are necessary for specific configs
+ */
+const __internal = {
```

Comment:
while this plugin is shipped to npm the intent is to not expose our internal APIs. without creating a new pipeline that would physically remove internal parts of code this should be a good enough solution for the temporary registration approach